### DISCUSSION TITLE
Covering auto-generated files not checked into git

### DISCUSSION THREAD

**Discussion by elringus** [codecov/feedback#125] [Ideas]
**What product do you want to improve?**
Not sure. Probably both Uploader and Web application.

**Is your feature request related to a problem? Please describe.**
I’m covering code generated on build, which is not checked into git.

The code is not known at compile time and has to be generated on build (hence it can’t be checking in git), but still needs to be tested and covered; I believe that’s a fair use case.

Everything works fine locally and on CI: I can test and generate coverage report for the generated code (vitest command: https://github.com/Elringus/DotNetJS/blob/feat/revamp/JavaScript/package.json#L4). But codecov ignores the report, because the covered files are not in git.

**Describe the solution you'd like**
Codecov will pick coverage for files not checked with git, respect their coverage in the context of the total solution coverage and display the files on the webapp. 

The non-git files may be marked as virtual/generated and don't have link to github repo.

**Describe alternatives you've considered**
My current workaround is to throw an error when coverage is below 100%, but I really wish codecov could support generated files.

**Additional context**
Migrated from https://community.codecov.com/t/covering-auto-generated-files-not-checked-into-git/4508

**Comment by rohan-at-sentry** [comment]
@Elringus thanks for raising this! I'm converting this issue to a discussion so as to bring this issue to a broader audience so we can collect feedback on this

**Comment by mtl1979** [comment]
I do think CodeCov should allow coverage reports for generated files, be it from a template file or generated by a script as long as the file name of the generated file is consistent between commits.

For templates, this would mean source files in repository having for example ".in" extension but the equivalent generated file not having it and located inside the build directory.

In both cases CodeCov should use the output filename to distinguish separate files.
